### DISCUSSION TITLE
Can't successfully render URI sprite

### DISCUSSION THREAD

**Discussion by UndarkAido** [pixijs/pixijs#8964] [Q&A] [answered]
My code, mostly the default example from the README, is as follows:
```js
            const app = new PIXI.Application();
            document.body.appendChild(app.view);

            //const uri = await readData64URIEntry(entryMap['assets/minecraft/textures/block/cobblestone.png'])
            //console.log('data:image/png;' + uri)
            const texture = await PIXI.Assets.load("data:image/png;data:;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAA0ElEQVQ4y22S0Q2FMAwDMyMbMABvjcck3YD5QKl01WHxUVDcxElc1+847n3f7zHGPNu2zRjsuq75P//nxPpvrJzYQJ8m6diJ4JB04xdBB/33RBQaT5JiXICMjUPGOk1W/XECYxNDCM7hrvqiCVIHjwxmTZYGqbB390o0Md6E5WRYc2+vRQ53ZTbUzmJIrQnalYtTeU9Ed686Rcyi7EKhReUllgZO8J7e3S+CS5cGTk5N0okWeVk5PW8rf73My8p2Vlr3q9i+mSKmZb9I3DVf7AEDw33Q7E89pwAAAABJRU5ErkJggg==");
            const block = new PIXI.Sprite(texture);

            block.x = app.renderer.width / 2;
            block.y = app.renderer.height / 2;

            block.anchor.x = 0.5;
            block.anchor.y = 0.5;

            app.stage.addChild(block);

            app.ticker.add(() => {
                block.rotation += 0.01;
            });
```
I just see a black rectangle with no errors in the console:
![image](https://user-images.githubusercontent.com/7421794/206828582-2eb6bc43-8a4a-4b3d-b707-12bb507885ed.png)
What am I doing wrong? The URI seems to be valid when I test it separately.

**Comment by SuperSodaSea** [comment, marked as answer]
A quick fix is to change your data URL to normal form `data:image/png;base64,iVBORw0KGgoAAAA...`

Although the data URL `data:image/png;data:;base64,iVBORw0KGgoAAAA...` looks strange, browsers consider it a valid data URL, but PixiJS doesn't. Maybe we need to modify `path.isDataUrl`.

https://github.com/pixijs/pixijs/blob/1325282801a17f1974bb9285d351435542257377/packages/utils/src/path.ts#L168-L173

**Reply by UndarkAido** [reply]
Hey, that did it! This was just my bad understanding of Base64 strings combined with not paying nearly enough attention. For anyone checking this out in the future here's `readData64URIEntry` to load an image from zip.js into Pixi:
```js
        async function readData64URIEntry(entry, mimeString=null) {
            const dataWriter = new zip.Data64URIWriter(mimeString)
            return await entry.getData(dataWriter)
        }
```
